src/plugins/workspace/public/components/workspace_form/workspace_use_case.tsx

```diff
@@ -54,21 +57,49 @@ const WorkspaceUseCaseCard = ({
   );
 };
 
+type AvailableUseCase = Pick<WorkspaceUseCaseObject, 'id' | 'title' | 'description' | 'systematic'>;
+
 export interface WorkspaceUseCaseProps {
   value: string | undefined;
   onChange: (newValue: string) => void;
   formErrors: WorkspaceFormErrors;
-  availableUseCases: Array<
-    Pick<WorkspaceUseCaseObject, 'id' | 'title' | 'description' | 'systematic'>
-  >;
+  availableUseCases: AvailableUseCase[];
+  savedObjects: SavedObjectsStart;
+  operationType: WorkspaceOperationType;
 }
 
 export const WorkspaceUseCase = ({
   value,
   onChange,
   formErrors,
   availableUseCases,
+  savedObjects,
+  operationType,
 }: WorkspaceUseCaseProps) => {
+  const [displayedUseCases, setDisplayedUseCases] = useState(availableUseCases);
+  const [isOnlyAllowEssential, setIsOnlyAllowEssential] = useState(false);
+
+  useEffect(() => {
+    if (operationType === WorkspaceOperationType.Create) {
+      getIsOnlyAllowEssentialUseCase(savedObjects.client).then((result: boolean) => {
+        setIsOnlyAllowEssential(result);
+      });
+    }
+  }, [savedObjects, operationType]);
+
+  useEffect(() => {
+    let allAvailableUseCases = availableUseCases
+      .filter((item) => !item.systematic)
+      .concat(DEFAULT_NAV_GROUPS.all);
+    // When creating and isOnlyAllowEssential is true, only display essential use case
+    if (isOnlyAllowEssential && operationType === WorkspaceOperationType.Create) {
+      allAvailableUseCases = allAvailableUseCases.filter(
+        (item) => item.id === DEFAULT_NAV_GROUPS.analytics.id
+      );
+    }
+    setDisplayedUseCases(allAvailableUseCases);
+  }, [availableUseCases, isOnlyAllowEssential, operationType]);
```

Comment:
The same here, we can simply use `useMemo`, isn't it?

Reply:
Updated.